src/propagate_constant.cpp

```diff
@@ -84,12 +84,12 @@ void propagate_constant::apply(module& m) const
     // Compute literals in parallel
     std::vector<instruction_ref> const_instrs_vec{const_instrs.begin(), const_instrs.end()};
     std::vector<argument> literals(const_instrs_vec.size());
-    std::size_t n = 1;
+    std::size_t grainsize = 1;
 #if !MIGRAPHX_HAS_EXECUTORS
-    n = std::max<std::size_t>(
-        std::ceil(static_cast<double>(1024) / std::thread::hardware_concurrency()), 1);
+    std::size_t n = std::max<std::size_t>(2048 / std::thread::hardware_concurrency(), 1);
+    grainsize     = const_instrs_vec.size() / n;
```

Comment:
Can this grainsize be zero? The earlier code seemed to avoid that case.. Thanks.

Reply:
Yes, the grainsize can be zero for the case `const_instrs_vec.size() == 0`. This, however, should not cause any issues. The threadsize in `simple_par_for` has been calculated to zero when `n(const_instrs_vec.size()) == 0`, even before. .
```
    const auto threadsize = std::min<std::size_t>(std::thread::hardware_concurrency(),
                                                  n / std::max<std::size_t>(1, min_grain));
```